test/Microsoft.SqlTools.ServiceLayer.UnitTests/Profiler/XeStreamObservableTests.cs

```diff
@@ -0,0 +1,92 @@
+﻿//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+#nullable disable
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using NUnit.Framework;
+using System.Reflection;
+using Microsoft.SqlTools.ServiceLayer.Profiler.Contracts;
+using Microsoft.SqlTools.ServiceLayer.Profiler;
+using System.Threading;
+using System.Linq;
+
+namespace Microsoft.SqlTools.ServiceLayer.UnitTests.Profiler
+{
+    public class XeStreamObservableTests
+    { 
+
+        /// <summary>
+        /// this might technically be an integration test but putting it here because it doesn't require any connectivity.
+        /// </summary>
+        [Test]
+        public void XeStreamObservable_reads_entire_xel_file()
+        {
+            var observer = InitializeFileObserver();
+            observer.Observable.Start();
+            var retries = 100;
+            while (!observer.Completed && retries-- > 0)
+            {
+                Thread.Sleep(100);
+            }
+            Assert.That(observer.Completed, Is.True, $"Reading the file didn't complete in 10 seconds. Events read: {observer.ProfilerEvents.Count}");
+            Assert.Multiple(() =>
+            {
+                Assert.That(observer.ProfilerEvents.Count, Is.EqualTo(149), "Number of events read");
+                Assert.That(observer.ProfilerEvents[0].Name, Is.EqualTo("rpc_completed"), "First event in the file");
+                Assert.That(observer.ProfilerEvents.Last().Name, Is.EqualTo("sql_batch_completed"), "Last event in the file");
+
+            });
+        }
+
+        [Test]
+        public void XeStreamObservable_calls_OnError_when_the_fetcher_fails()
+        {
+            var observer = InitializeFileObserver("thispathdoesnotexist.xel");
+            Assert.Throws<FileNotFoundException>(observer.Observable.Start);    // exception handled in StartLocalFileSession task (ProfilerServic.cs)

```

Comment:
>tion>(observer.Observable.Start);  [](http://example.com/codeflow?start=43&length=34)

this behavior is not aligned with the Observable contract.
`Start` should always return an IObservable instance and not throw .

Reply:
Addressed in latest iteration.